### ISSUE TITLE
Happy, thanks

### ISSUE THREAD

**Issue by EdKok** [pacmanmt/jkbms#1]
Thanks for sharing.
Took a bit of time to get it working, but it does!

Had to create the var $(serport) and add an inject node to get the flow started. Will amend to my liking but already very happy.

Would you happen to know how to enable/disable AllowToCharge, discharge and Balance?

**Comment by pacmanmt** [comment]
Let me know if you need anything else, glad to help if I can.
With the firmware version my BMSs have, you can drive only the charge and discharge switches. The balance one has to be operated from the app.
JK justified this for "security reasons"...😲

**Comment by EdKok** [comment]
@pacmanmt Actually "Balance" isn't important to be able to switch remotely, but "Charge" would be great. Do you know how to do that with Node-Red?

Reason: My battery is on a boat, being charged with a (smallish) alternator that could overheat. The temperature of the alternator is being monitored so could potentially trigger the "Charge" to off with NR.

**Comment by pacmanmt** [comment]
Just added the FETs toggle subflow.

**Comment by pacmanmt** [comment]
Hi EdKok,
have you been able to integrate the FETs flow?

**Comment by EdKok** [comment]
Yes, today. Did a bit of rework and changes because I’m not sending to mqtt but to “Signal K”, an open-source marine project. The flow for switching the FET’s is now integrated by setting a flow variable and the first function in your flow looks for this var and decides om what msg to send. Either the original msg to read all the metrics, or send one of the four strings to switch the FET’s. The vars are being set by a marine touch display with virtual switches. All working; very happy. All thanks to your work and willingness to share. So thanks again.

**Comment by EdKok** [comment]
![FB8D368E-2378-4F4E-A7BB-66DE6ED05097](https://github.com/user-attachments/assets/e0784a79-9c87-44e7-861b-3e5f5a237d91)

**Comment by pacmanmt** [comment]
Really glad the flows helped you to complete the integration. 👍
